dali/operators/image/resize/resize_op_impl_cpu.h

```diff
@@ -0,0 +1,131 @@
+// Copyright (c) 2020, NVIDIA CORPORATION. All rights reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef DALI_OPERATORS_IMAGE_RESIZE_RESIZE_OP_IMPL_CPU_H_
+#define DALI_OPERATORS_IMAGE_RESIZE_RESIZE_OP_IMPL_CPU_H_
+
+#ifndef DALI_RESIZE_BASE_CC
+#error This file is a part of resize base implementation and should not be included elsewhere
+#endif
+
+#include <cassert>
+#include <cmath>
+#include <vector>
+#include "dali/operators/image/resize/resize_op_impl.h"
+#include "dali/kernels/imgproc/resample_cpu.h"
+
+namespace dali {
+
+template <typename Out, typename In, int spatial_ndim>
+class ResizeOpImplCPU : public ResizeBase<CPUBackend>::Impl {
+ public:
+  explicit ResizeOpImplCPU(kernels::KernelManager &kmgr, int num_threads) : kmgr_(kmgr) {
+    kmgr_.Resize(num_threads, 0);
+  }
+
+  static_assert(spatial_ndim == 2, "NOT IMPLEMENTED. Only 2D resize is supported");
+
+  using Kernel = kernels::ResampleCPU<Out, In, spatial_ndim>;
+
+  /// Dimensionality of each separate frame. If input contains no channel dimension, one is added
+  static constexpr int frame_ndim = spatial_ndim + 1;
+
+  void Setup(TensorListShape<> &out_shape,
+             const TensorListShape<> &in_shape,
+             int first_spatial_dim,
+             span<const kernels::ResamplingParams> params) override {
+    // Calculate output shape of the input, as supplied (sequences, planar images, etc)
+    GetResizedShape(out_shape, in_shape, params, spatial_ndim, first_spatial_dim);
+
+    // Create "frames" from outer dimensions and "channels" from inner dimensions.
+    GetFrameShapesAndParams<spatial_ndim>(in_shape_, params_, in_shape, params,
+                                          first_spatial_dim);
+
+    // Now that we have per-frame parameters, we can calculate the output shape of the
+    // effective frames (from videos, channel planes, etc).
+    GetResizedShape(out_shape_, in_shape_, make_cspan(params_), 0);
+
+    // Now that we know how many logical frames there are, calculate batch subdivision.
+    OnNumFramesUpdated();
+
+    SetupKernel();
+  }
+
+  void SetupKernel() {
+    const int dim = in_shape_.sample_dim();
+    kernels::KernelContext ctx;
+
+    for (int i = 0; i < GetNumFrames(); i++) {
+      kernels::InTensorCPU<In, frame_ndim> dummy_input;
```

Comment:
:(

Reply:
I guess I'll redo it when dealing with 3D resize - but I want to avoid merging the changes now.